run.py

```diff
@@ -68,8 +72,13 @@
             explosion_group.add(Explosion(target.rect.centerx, target.rect.centery, target.rect.size))
             score_group.add(Score(target.points, target.rect.width, target.rect.center, display.get_rect()))
             player.update_score(target.points)
-            target_group.add(target.spawn_new_target(target_group, player))
-
+            target_group.add(Target())
+        #checks for player tank / target collision. removes target with no explosion and no points given
+        hits = pygame.sprite.groupcollide(target_group, player_group, True, False)
+        for target in hits:
+            target_group.add(Target())
```

Comment:
I don't think you need a for loop here.  There's ever only one target on the field at once, so if you've killed it upon collision, then add a new one to the group.

Reply:
true, maybe just put the group collide in an if statement and add if the condition is met? that seems to better